client/src/main/scala/org/http4s/client/middleware/FollowRedirect.scala

```diff
@@ -65,8 +65,10 @@ object FollowRedirect {
         // We can only resubmit a body if it was not effectful.
         def pureBody = {
           req.body.unemit match {
-            case (chunks, _) if chunks.nonEmpty => Some(chunks.reduce(_ ++ _))
-            case _ => None
+            case (chunks, p) if chunks.nonEmpty && p.isHalt =>
```

Comment:
Well I agree with that: we don't want to reduce an empty collection. :smile: But what I mean is, why can't this be
```scala
req.body.unemit match {
  case (chunks, p) p.isHalt =>
    Some(chunks.fold(ByteVector.empty)(_ ++ _))

  case _ =>
    None
}
```
to allow redirect on a PUT request with an empty body.